Nasal/light-manager.nas

```diff
@@ -0,0 +1,303 @@
+        delta_y = -(apos.lon() - vpos.lon()) * me.lon_to_m;
+        delta_z = apos.alt() - vpos.alt();
+
+        setprop("/sim/rendering/als-secondary-lights/lightspot/eyerel-x-m[3]", delta_x);
+        setprop("/sim/rendering/als-secondary-lights/lightspot/eyerel-y-m[3]", delta_y);
+        setprop("/sim/rendering/als-secondary-lights/lightspot/eyerel-z-m[3]", delta_z);
+
+        settimer(func me.update(), 0.0);
+    },   
+
+    light1_on: func {
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-r", me.light1_r);
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-g", me.light1_g);
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-b", me.light1_b);
+    },
+
+    light1_off: func {
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-r", 0.0);
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-g", 0.0);
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-b", 0.0);
+    },
+
+    light2_on: func {
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-r[1]", me.light2_r);
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-g[1]", me.light2_g);
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-b[1]", me.light2_b);
+    },
+
+    light2_off: func {
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-r[1]", 0.0);
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-g[1]", 0.0);
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-b[1]", 0.0);
+    },
+
+    light3_on: func {
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-r[2]", me.light3_r);
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-g[2]", me.light3_g);
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-b[2]", me.light3_b);
+    },
+
+    light3_off: func {
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-r[2]", 0.0);
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-g[2]", 0.0);
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-b[2]", 0.0);
+    },
+
+    light4_on: func {
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-r[3]", me.light4_r);
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-g[3]", me.light4_g);
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-b[3]", me.light4_b);
+    },
+
+    light4_off: func {
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-r[3]", 0.0);
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-g[3]", 0.0);
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-b[3]", 0.0);
+    },
+    
+    enable_or_disable: func (enable, light_num) {
+        if (enable) {
```

Comment:
I considered doing something like what you suggested above but got lazy and just plugged in the function as is. I should have known yu would call me on it! :smile: 

I changed it to the following. I didn't do any arrays as per your example, again. Instead I did the following.
~~~xml
    switch_position: func(light, lightr, lightg, lightb) {
        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-r["~light~"]", lightr);
        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-g["~light~"]", lightg);
        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-b["~light~"]", lightb);
    },

    enable_or_disable: func (enable, light_num) {
        if (enable) {
            if (light_num == 0)
                me.switch_position(light_num, me.light1_r, me.light1_g, me.light1_b);
            if (light_num == 1)
                me.switch_position(light_num, me.light2_r, me.light2_g, me.light2_b);
            if (light_num == 2)
                me.switch_position(light_num, me.light3_r, me.light3_g, me.light3_b);
            if (light_num == 3)
                me.switch_position(light_num, me.light4_r, me.light4_g, me.light4_b);
        } else {
            me.switch_position(light_num, 0.0, 0.0, 0.0);
        }
    },
~~~